python_modules/dagster/dagster/_core/asset_graph_view/asset_graph_view.py

```diff
@@ -142,6 +157,26 @@ def compute_intersection_with_partition_keys(
         """Return a new AssetSlice with only the given partition keys if they are in the slice."""
         return self._asset_graph_view.compute_intersection_with_partition_keys(partition_keys, self)
 
+    @property
+    def time_windows(self) -> Sequence[TimeWindow]:
+        """Get the time windows for the asset slice. Only supports explicitly time-windowed partitions for now."""
+        # Only supports explicitly time-windows partitions for now
```

Comment:
I'm going to change this to support all time windows later. In a bit of rebase hell right now.